sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java

```diff
@@ -679,6 +714,20 @@ private static Object convertAvroArray(
     return ret;
   }
 
+  private static Object convertAvroMap(
+      FieldType beamField, Object value, BigQueryUtils.ConversionOptions options) {
+    List<GenericData.Record> records = (List<GenericData.Record>) value;
+    HashMap<Object, Object> ret = new HashMap<>();
```

Comment:
We prefer to use ImmutableMap when possible, there are examples in this file.

Reply:
Sure, I updated it to use ImmutableMap instead.